roles/keystone/templates/etc/apache2/sites-available/keystone-admin.conf

```diff
@@ -0,0 +1,17 @@
+<VirtualHost *:35358>
+    LogLevel  info
+    ErrorLog  /var/log/apache2/keystone-error.log
+    CustomLog /var/log/apache2/keystone-access.log combined
+    Options +FollowSymLinks
+    ProxyPass / balancer://keystone
+
+    Include /etc/apache2/ssl.conf
+    SSLCertificateFile /etc/apache2/ssl.crt
+    SSLCertificateKeyFile /etc/apache2/ssl.key
+</VirtualHost>
+
+<Proxy balancer://keystone>
+{% for host in groups['controller'] -%}
+  BalancerMember http://{{ hostvars[host][primary_interface]['ipv4']['address'] }}:35357/
```

Comment:
Shouldn't this be 35358? or do we have keystone locally listening on 35357 (the default), and only front end 35358 in apache?